Source/WebCore/rendering/svg/SVGTextMetricsBuilder.cpp

```diff
@@ -76,9 +76,27 @@ void SVGTextMetricsBuilder::advanceSimpleText()
 void SVGTextMetricsBuilder::advanceComplexText()
 {
     unsigned metricsLength = currentCharacterStartsSurrogatePair() ? 2 : 1;
+
+    // FIXME: NON-COCOA platforms shifts some pixels for SVGText when x,y, and rotate are used.
+    // webkit.org/b/291636
+#if PLATFORM(COCOA)
+    float beforeWidth = 0;
+    float afterWidth = 0;
+
+    m_complexTextController->advance(m_textPosition, nullptr);
+    beforeWidth = m_complexTextController->runWidthSoFar();
+
+    m_complexTextController->advance(m_textPosition + metricsLength, nullptr);
+    afterWidth = m_complexTextController->runWidthSoFar();
+
+    m_currentMetrics = SVGTextMetrics(*m_text, metricsLength, afterWidth - beforeWidth);
+    m_complexStartToCurrentMetrics = SVGTextMetrics(*m_text, m_textPosition + metricsLength, afterWidth);
+
+    ASSERT(m_currentMetrics.length() == metricsLength);
+#else
     m_currentMetrics = SVGTextMetrics::measureCharacterRange(*m_text, m_textPosition, metricsLength);
     m_complexStartToCurrentMetrics = SVGTextMetrics::measureCharacterRange(*m_text, 0, m_textPosition + metricsLength);
-    ASSERT(m_currentMetrics.length() == metricsLength);
```

Comment:
I guess we have no reason for deleting this ASSERT for the other ports

Reply:
added back